delta-app/src/main/java/io/cdap/delta/store/AssessmentServiceClient.java

```diff
@@ -96,37 +92,41 @@ public AssessmentServiceClient(ServiceDiscoverer context) {
    * Copied from "AbstractServiceRetryableMacroEvaluator"
    * Returns the full content of the error stream for the given {@link HttpURLConnection}.
    */
-  private <T> T validateAndRetrieveContent(@Nullable HttpURLConnection urlConn, Class<T> type) throws IOException {
+  private <T> T validateAndRetrieveContent(@Nullable HttpURLConnection urlConn, Type type) throws IOException {
     String serviceName = "State Store (AssessorService)";
     if (urlConn == null) {
       throw new RetryableException(serviceName + " service is not available");
     }
 
+    if (type == null) {
+      throw new NullPointerException(" Type parameter cannot be null");
+    }
+
     int responseCode = urlConn.getResponseCode();
     if (responseCode != HttpURLConnection.HTTP_OK) {
       throw new RetryableException("Failed to call " + serviceName + " service with status " + responseCode + ": " +
                                      getError(urlConn));
     }
 
     try (InputStream inputStream = urlConn.getInputStream()) {
-      String contentType = urlConn.getContentType();
-      switch (contentType) {
-        case MediaType.APPLICATION_JSON: {
-          return GSON.fromJson(new InputStreamReader(inputStream, StandardCharsets.UTF_8), type);
-        }
-
-        case MediaType.APPLICATION_OCTET_STREAM: {
-          return (T) ByteStreams.toByteArray(inputStream);
-        }
-
-        default:
-          return (T) CharStreams.toString(new InputStreamReader(inputStream, StandardCharsets.UTF_8));
-      }
+      return type == byte[].class ?  (T) retrieve(inputStream) :  retrieve(inputStream, type);
```

Comment:
should not be type checking here, there should be a method that returns a byte[] and doesn't take any Type. The one that takes a Type and returns T should always be for a json response (technically you can represent a byte[] in json as a literal json array of numbers).

Think of the byte[] as the method used to get the raw byte content of the response, while the Type one is used to get and decode json response.